code/modules/events/apc_short.dm

```diff
@@ -27,51 +27,105 @@
 	return TRUE
 
 /proc/power_failure(announce=TRUE)
```

Comment:
Careful here; this proc is called by the event code on Line 15 above.
I'm not sure what value an `alert()` will return when not directly invoked by a client admin verb.
Or to put it another way: When the server fires the proc as an event, who would get the pop-up selection dialog to choose what happens?

Test that the event still works by opening the Admin -> Event Manager Panel, choosing APC Short for the Moderate event, then reduce the time in order to make the event fire within in a few seconds.

APC Short does not correct itself when the event is over, so the changes to the `power_restore` proc shouldn't affect the event.

Reply:
@blinkdog Thank goodness I keyed you in on this. You are a lifesaver. I'm going to have to test that and I'm expecting one of two outcomes:
A.) Server crashes
B.) The alert pops up
Either way, that's _not_ what we want.
The solution would be to plug in the alert somewhere else, probably `topics.dm`, and make two separate procs:
- `power_failure` would be your APC shorting code
- `apc_depower` would be the other one

Reply:
Hypotheses are great, especially when they are wrong.

> A.) Server crashes
> B.) The alert pops up

No, let's try:

C.) No alert pops up and it defaults to "Depower all APCs"